crates/sui-types/src/sui_serde.rs

```diff
@@ -241,3 +241,35 @@ impl<'de> DeserializeAs<'de, AggregateAuthoritySignature> for AggrAuthSignature
             .map_err(to_custom_error::<'de, D, _>)
     }
 }
+
+#[macro_export]
+macro_rules! serde_to_from_bytes {
+    ($type:ty) => {
+        impl ::serde::Serialize for $type {
+            fn serialize<S: ::serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+                match serializer.is_human_readable() {
+                    true => serializer.serialize_str(&self.encode_base64()),
+                    false => self.as_ref().serialize(serializer),
```

Comment:
this one adds the length as a prefix, can you use the macro we have in fastcrypto?

Reply:
the limitation of the serialize_deserialize_to_from_bytes is, it requires a known length. here a generic sig has unknown length for either multisig or a single sig (different schemes has different length)